### ISSUE TITLE
E3002 'DeletionProtection' was unexpected for EKS Cluster

### ISSUE THREAD

**Issue by txynidakis** [aws-cloudformation/cfn-lint#4228]
### CloudFormation Lint Version

1.39.0

### What operating system are you using?

Mac/Amazon Linux 2024

### Describe the bug

Hi

We use cnf-lint with pre-commit.

We're getting the error. 
E3002 Additional properties are not allowed ('DeletionProtection' was unexpected).
This relates to the EKS Cluster resource.
The deletion protection is a recent update to the EKS Cluster resource
https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-eks-cluster.html#cfn-eks-cluster-deletionprotection

Could you update the code in the next release?

Kind regards
Tony

### Expected behavior

The pre-commit hook will pass the CFN linting.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09
Description: Template to build EKS Cluster and Nodes 
Parameters:
  EnvironmentValue:
    AllowedValues:
      - Model
      - Production
    Default: Model
    Type: String
  BuildNumber:
    Type: String
  VPCStackName:
    Type: String
    Default: ""
    Description: Name of the CloudFormation Stack that created the VPC this cluster is being deployed into, e.g. VPC-Test
  EKSVersion:
    Description: What version of EKS to run
    Type: String
    AllowedPattern: '1.\d\d'

Resources:
  EKSCluster:
    Type: "AWS::EKS::Cluster"
    Properties:
      AccessConfig:
        AuthenticationMode: API_AND_CONFIG_MAP
      DeletionProtection: true
      Name: !Join
        - ""
        - - !Ref "AWS::StackName"
          - "-"
          - !Ref BuildNumber
      Version: !Ref EKSVersion
      RoleArn: !GetAtt
        - EKSServiceRole
        - Arn
      ResourcesVpcConfig:
        EndpointPublicAccess: false
        EndpointPrivateAccess: true
        SecurityGroupIds:
          - !Ref EKSControlPlaneSecurityGroup
        SubnetIds:
          - "Fn::ImportValue": !Sub "${VPCStack}-SubnetEKSAZa"
          - "Fn::ImportValue": !Sub "${VPCStack}-SubnetEKSAZb"
          - "Fn::ImportValue": !Sub "${VPCStack}-SubnetEKSAZc"

```

**Comment by kddejong** [comment]
Make sure you are actually using cfn-lint 1.39.0 as this is fixed in this release or do you run in a different region?
